---
title: "Figures 1B, Fig2A,B,C and Fig4B,C"
output:
  word_document: default
  html_notebook: default
---



```{r}
library(reshape2)
library(ggplot2)
library(tidyverse)
library(gridExtra)
library(tidyr)
library(ggpubr)
library("readxl")
library(datasets)  
library(purrr)
library(ggridges)
```
-Figure 1.B 'we noticed that DEGs are more transcriptionally isolated then SEGs'
```{r}
read_tsv("data_figures/DEG_SEG_dist_thrE.txt")->DEG_SEG_dist_thr

DEG_SEG_dist_thr%>%
    ggplot(aes(set_ds, log.dist, fill = set_ds)) +
    geom_violin(draw_quantiles = 0.5) +
    theme_bw(base_size = 16) +
    scale_fill_manual(values = c("blue","yellow")) +
        ylab("dist2nearest expr gene (log10 bp)\n") +
    xlab(element_blank()) +
    theme(axis.title.y = element_text(size = 14), plot.title = element_text(size = 14, hjust = 0.5)) + 
    ggtitle("dist for DEG SEG") -> v2ds.deg
  v2ds.deg #all thr dumpted together

v2ds.deg+facet_grid(~thrGE)+ theme(legend.position = "none")+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5)) ->v02
v02+ggtitle("distances in  DEG SEG per GE threshold")
#------------------------no title
v02+ggtitle("")
```
-Figure 2. A B C DEG vs SEG scatter EG and densities, pairwise here
'show how we computed DEGs and SEGs'
```{r}
#---------------------Fig2.A
read_tsv("data_figures/pw_DEG_histo.txt")->diff
#head(diff)

pdiff<- ggplot(diff, aes (x=GE_diff, y=freq, fill=abs(GE_diff))) +
  geom_bar(stat='identity',show.legend = FALSE)+
  scale_colour_gradient2()+
  theme_minimal(base_size = 14)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  #ggtitle("A or C")+
  xlab("Gene expression difference")+
  ylab("counts")
pdiff

```
-Fig2 B `scatter plots for SEG,  Dect Dmes

```{r}
read_tsv("data_figures/scat_sim.txt")->scat_sim
#head(scat_sim)

read_tsv("data_figures/scat_ect.txt")->scat_ect
#head(scat_ect)

read_tsv("data_figures/scat_mes.txt")->scat_mes
#head(scat_mes)

pp<- ggplot(scat_sim, aes (x=ge_ect,y=ge_mes)) +
  geom_point(size=2,color="orange")+
  geom_point(data = scat_ect,color="green",size=2)+
  geom_point(data = scat_mes,color="magenta",size=2)+
  ylim(-10, 12)+
  theme_minimal(base_size = 14)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  #ggtitle("C")+
  xlab("Gene expression Ect")+
  ylab("Gene expression Mes")
pp
```
-Fig2. C
```{r}

read_tsv("data_figures/GE_long_SEG.txt")->Sim


pd4<- ggplot(Sim, aes(x = GE, y = layer,fill=layer)) +    geom_density_ridges(
       aes(point_fill = layer, point_shape = layer),
       alpha = .4) +
       xlab("Gene expression")+
       ylab(" ")+
       scale_fill_manual(values = c("green", "blue","magenta"))
#pd4

#-------------------------------all genes

read_tsv("data_figures/GE_long_ALL.txt")->ALL


pd1<- ggplot(ALL, aes(x = v, y = layers,fill=layers)) +    geom_density_ridges(
       aes(point_fill = layers, point_shape = layers),
       alpha = .4) +
       xlab(" ")+
       ylab(" ")+
       scale_fill_manual(values = c("green", "blue","magenta"))
 #, labels = c("female", "male","unknown")) 
#pd1

#---------------------DifEct
read_tsv("data_figures/GE_long_DifEct.txt")->DifEct

pd2<- ggplot(DifEct, aes(x = GE, y = layer,fill=layer)) +    geom_density_ridges(
       aes(point_fill = layer, point_shape = layer),
       alpha = .4) +
       xlab(" ")+
       ylab(" ")+
       scale_fill_manual(values = c("green", "blue","magenta"))
 #, labels = c("female", "male","unknown")) 
#pd2

#--------------------Dif Mes
read_tsv("data_figures/GE_long_DifMes.txt")->DifMes

pd3<- ggplot(DifMes, aes(x = GE, y = layer,fill=layer)) +    geom_density_ridges(
       aes(point_fill = layer, point_shape = layer),
       alpha = .4) +
       xlab("Gene expression")+
  ylab(" ")+
       scale_fill_manual(values = c("green", "blue","magenta"))
 #, labels = c("female", "male","unknown")) 
#pd3
```
Fig2 C: arrange densities together
```{r}
pE <-
  ggarrange( pd1, pd2, pd3, pd4, labels = c("C1","C2","C3", "C4"),
           common.legend = TRUE, legend = "bottom"
  )
pE
```

-Figure 4.B Distribution of chromatin accessibility difference in a window along chromosome 1.

-DAR density in 100bp window, Ect vs Mes, chr1 = Fig4.B

```{r}
read_tsv("data_figures/da_win_pw_chr1.txt")->da_win

#head(da_win)

pDAR<-  ggplot(da_win, aes(diff_access))+
  #geom_bar(stat='identity')
  geom_density(fill='grey')+
  #ggtitle("Differential access,100bp win       Ect vs Meso")+
  theme_minimal(base_size = 14)+
  ylab("density")+
  xlab("differential access, frac")
pDAR

```
-Figure 4.C TSS histograms

```{r}
read_tsv("data_figures/histTSS_DEG.txt")->histTSS_DEG
head(histTSS_DEG)

#-------------------histo SEG

ps <- ggplot(histTSS_DEG, aes(x=TSS_vic, y=freqSEG, colour=type_SEG)) + 
    geom_line(size=1)+
    scale_colour_manual(values=c("black","orange","yellow"))+
    
    theme_minimal(base_size=15)+
    ylim(0,0.6)+
     #ylab("normalised chromatin access count")+
    xlab(" ") + 
    ylab(" ")+
   ylab("   Normalised chromatin access")+
     #ylab("Normalised per gene Chromat Access count ")+
    theme(legend.title = element_blank())
  
  ps<-ps+scale_x_continuous(breaks=seq(-80000,80000,80000),labels=c("-80K", "TSS", "80K"))
  
  #ps
  
  #pp<-grid.arrange(pe, pm, ps, ncol=1)
  #pp + theme(legend.title = element_blank())
  
  #---------------------DEG TSS
  
pm <- ggplot(histTSS_DEG, aes(x=TSS_vic, y=freqDEG, colour=type_DEG))+ geom_line(size=1)+
  scale_colour_manual(values=c("black","blue","cyan"))+
  #xlim(-80000,80000)+
  theme_minimal(base_size=14)+
  xlab(" ") + 
  ylab(" ")+
 #ylab("Normalised Chromatin Access")+
  scale_y_continuous(breaks=seq(-0,0.6,0.2))+
  theme(legend.title = element_blank())
#pm
    pm<-pm+scale_x_continuous(breaks=seq(-80000,80000,80000),labels=c("-80K", "TSS","80K"))
  
  #--------------------------together
  
  pp<-grid.arrange(pm, ps, ncol=1)
pp

```


