---
title: "Figures 1B, Fig2A,B,C, Fig3 B, C, and Fig4B,C, Fig6A, B"
output:
  word_document: default
  html_notebook: default
---



```{r}
library(reshape2)
library(ggplot2)
library(tidyverse)
library(gridExtra)
library(tidyr)
library(ggpubr)
library("readxl")
library(datasets)  
library(purrr)
library(ggridges)
library(UpSetR)
library(VennDiagram)

```
-Figure 1.B 'we noticed that DEGs are more transcriptionally isolated then SEGs'
```{r}
read_tsv("data_figures/DEG_SEG_dist_thrE.txt")->DEG_SEG_dist_thr

DEG_SEG_dist_thr%>%
    ggplot(aes(set_ds, log.dist, fill = set_ds)) +
    geom_violin(draw_quantiles = 0.5) +
    theme_bw(base_size = 16) +
    scale_fill_manual(values = c("blue","yellow")) +
        ylab("dist2nearest expr gene (log10 bp)\n") +
    xlab(element_blank()) +
    theme(axis.title.y = element_text(size = 14), plot.title = element_text(size = 14, hjust = 0.5)) + 
    ggtitle("dist for DEG SEG") -> v2ds.deg
  v2ds.deg #all thr dumpted together

v2ds.deg+facet_grid(~thrGE)+ theme(legend.position = "none")+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.5)) ->v02
v02+ggtitle("distances in  DEG SEG per GE threshold")
#------------------------no title
v02+ggtitle("")
```
-Figure 2. A B C DEG vs SEG scatter EG and densities, pairwise here
'show how we computed DEGs and SEGs'
```{r}
#---------------------Fig2.A
read_tsv("data_figures/pw_DEG_histo.txt")->diff
#head(diff)

pdiff<- ggplot(diff, aes (x=GE_diff, y=freq, fill=abs(GE_diff))) +
  geom_bar(stat='identity',show.legend = FALSE)+
  scale_colour_gradient2()+
  theme_classic2(base_size = 14)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  #ggtitle("A or C")+
  xlab("GE difference log2RPKM")+
  ylab("counts")
pdiff

```
-Fig2 B `scatter plots for SEG,  Dect Dmes

```{r}
read_tsv("data_figures/scat_sim.txt")->scat_sim
#head(scat_sim)

read_tsv("data_figures/scat_ect.txt")->scat_ect
#head(scat_ect)

read_tsv("data_figures/scat_mes.txt")->scat_mes
#head(scat_mes)

pp<- ggplot(scat_sim, aes (x=ge_ect,y=ge_mes)) +
  geom_point(size=2,color="orange")+
  geom_point(data = scat_ect,color="green",size=2)+
  geom_point(data = scat_mes,color="magenta",size=2)+
  ylim(-10, 12)+
  theme_classic2(base_size = 14)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())+
  #ggtitle("C")+
  xlab("GE Ectoderm log2RPKM")+
  ylab("GE Mesoderm")

pp<-pp+scale_x_continuous(breaks=seq(-10,10,10),labels=c("-10", "0", "10"))+scale_y_continuous(breaks=seq(-10,10,10),labels=c("-10", "0", "10"))
pp
```
-Fig2AB
```{r, fig.width = 8}
ppp <-
  ggarrange( pdiff, pp, labels = c("A","B"))
ppp
```

-Fig2. C
```{r}

read_tsv("data_figures/GE_long_SEG.txt")->Sim


pd4<- ggplot(Sim, aes(x = GE, y = layer,fill=layer)) +    geom_density_ridges(
       aes(point_fill = layer, point_shape = layer),
       alpha = .4, show.legend = F) +
       theme_classic2(base_size = 14)+
       #xlab("Gene expression")+
       xlab("GE, log2RPKM")+
       ylab(" ")+
       scale_fill_manual(values = c("green", "blue","magenta"))
#pd4
pd4<-pd4+scale_x_continuous(breaks=seq(-10,10,10),labels=c("-10", "0", "10"))

#-------------------------------all genes

read_tsv("data_figures/GE_long_ALL.txt")->ALL


pd1<- ggplot(ALL, aes(x = v, y = layers,fill=layers)) +    geom_density_ridges(
       aes(point_fill = layers, point_shape = layers),
       alpha = .4, show.legend = F) +
       theme_classic2(base_size = 14)+
       xlab(" ")+
       ylab(" ")+
       scale_fill_manual(values = c("green", "blue","magenta"))
 #, labels = c("female", "male","unknown")) 
#pd1
pd1<-pd1+scale_x_continuous(breaks=seq(-10,10,10),labels=c("-10", "0", "10"))

#---------------------DifEct
read_tsv("data_figures/GE_long_DifEct.txt")->DifEct

pd2<- ggplot(DifEct, aes(x = GE, y = layer,fill=layer)) +    geom_density_ridges(
       aes(point_fill = layer, point_shape = layer),
       alpha = .4,show.legend = F) +
       theme_classic2(base_size = 14)+
       xlab(" ")+
       ylab(" ")+
       scale_fill_manual(values = c("green", "blue","magenta"))
  
pd2<-pd2+scale_x_continuous(breaks=seq(-10,10,10),labels=c("-10", "0", "10"))
 
#pd2

#--------------------Dif Mes
read_tsv("data_figures/GE_long_DifMes.txt")->DifMes

pd3<- ggplot(DifMes, aes(x = GE, y = layer,fill=layer)) +    geom_density_ridges(
       aes(point_fill = layer, point_shape = layer),
       alpha = .4, show.legend = F) +
       theme_classic2(base_size = 14)+
       xlab("GE, log2RPKM")+
       ylab(" ")+
       scale_fill_manual(values = c("green", "blue","magenta"))
 
#pd3
pd3<-pd3+scale_x_continuous(breaks=seq(-10,10,10),labels=c("-10", "0", "10"))
```
Fig2 C: arrange densities together
```{r,fig.width=8}
#pE <-
  #ggarrange( pd1, pd2, pd3, pd4, labels = c("C1","C2","C3", "C4"),
           #common.legend = TRUE, legend = "bottom"
  #)
pE <-  ggarrange( pd1, pd2, pd3, pd4, labels = c("C1","C2","C3", "C4"))#+theme_classic2(base_size = 12)
pE
```
-Figure 3 TADs and genes

-Fig3C. TAD borders for DEGs and SEGs, data
```{r}

read_tsv("data_figures/DEG_SEG_TAD_borders.txt")->dsTAD
#head(dsTAD)

dsTAD %>%
  mutate(DEG=db/(db+dm)) %>%
  mutate(SEG=sm/(sb+sm)) %>%
  #made one long format out of two columns, their names are other long column of names 
gather(key=gene_set, value=frac,DEG,SEG) -> dsTAD1
#tail(dsTAD1)


```
--Fig3C. TAD borders for DEGs and SEGs, density distribution of fractions: DEG-TAD boders vs SEG TAD borders
```{r}
pdt<- ggplot(dsTAD1, aes(x = frac, y = gene_set,fill=gene_set)) +    geom_density_ridges(
       aes(point_fill = gene_set, point_shape = gene_set),
       alpha = .4) +
       xlab("fraction of genes near TAD borders")+
       ylab(" ")+
 
       scale_fill_manual(values = c("blue", "orange"))
 pdt

```
- fraction of gene-TAD single, double etc TADs : DEG SEG

```{r}

read_tsv("data_figures/frac_DEG_SEG_TAD_long2.txt")->fr_ds_TADL2
head(fr_ds_TADL2)



pb2<- ggplot(fr_ds_TADL2, aes(x=counts,y=frac,fill=set_TAD))+
  geom_bar(stat='identity',show.legend = FALSE)+
       #geom_point(size=2)+
       theme_classic2(base_size = 14)+
       xlab("number of genes in a TAD ")+
       ylab("fraction of TADs ")+
       scale_fill_manual(values = c("blue", "orange"))
 pb2

pb2+facet_grid(~set_TAD)

```
-Figure 4.B Distribution of chromatin accessibility difference in a window along chromosome 1.

-DAR density in 100bp window, Ect vs Mes, chr1 = Fig4.B

```{r}
read_tsv("data_figures/da_win_pw_chr1.txt")->da_win

#head(da_win)

pDAR<-  ggplot(da_win, aes(diff_access))+
  #geom_bar(stat='identity')
  geom_density(fill='grey')+
  #ggtitle("Differential access,100bp win       Ect vs Meso")+
  theme_minimal(base_size = 14)+
  ylab("density")+
  xlab("differential access, frac")
pDAR

```
-Figure 4.C TSS histograms

```{r}
read_tsv("data_figures/histTSS_DEG.txt")->histTSS_DEG
head(histTSS_DEG)

#-------------------histo SEG

ps <- ggplot(histTSS_DEG, aes(x=TSS_vic, y=freqSEG, colour=type_SEG)) + 
   #geom_bar(stat='identity',show.legend = FALSE)+

    geom_line(size=1,show.legend = FALSE)+
    scale_colour_manual(values=c("black","orange","yellow"))+
    
    theme_minimal(base_size=14)+
    ylim(0,0.6)+
     #ylab("normalised chromatin access count")+
    xlab(" ") + 
    ylab(" ")+
   ylab("              Chromatin Accessibility")+
     #ylab("Normalised per gene Chromat Access count ")+
    theme(legend.title = element_blank())
  
  ps<-ps+scale_x_continuous(breaks=seq(-80000,80000,80000),labels=c("-80K", "TSS SEGs", "80K"))
  
  #ps
  
  #pp<-grid.arrange(pe, pm, ps, ncol=1)
  #pp + theme(legend.title = element_blank())
  
  #---------------------DEG TSS
  
pm <- ggplot(histTSS_DEG, aes(x=TSS_vic, y=freqDEG, colour=type_DEG))+ geom_line(size=1,show.legend = FALSE)+
  scale_colour_manual(values=c("black","blue","cyan"))+
  #xlim(-80000,80000)+
  theme_minimal(base_size=14)+
  xlab(" ") + 
  ylab(" ")+
 #ylab("Chromatin Accessibility")+
  scale_y_continuous(breaks=seq(-0,0.6,0.2))+
  theme(legend.title = element_blank())
#pm
    pm<-pm+scale_x_continuous(breaks=seq(-80000,80000,80000),labels=c("-80K", "TSS DEGs","80K"))
  
  #--------------------------together
  
  pp<-grid.arrange(pm, ps, ncol=1)
pp

```

Fig 4.D Left
```{r}
read_tsv("data_figures/DAR_aroundH3K27ac_Mes_20K.txt")->dar_enh

pmd<- ggplot(dar_enh, aes(bin,line))+
  geom_line(color="blue",size=1) +
  ylim(0, 200)+
 #ggtile("DARM normalised counts around H3K27acMes") 
  
  xlab(" ")+
  ylab("normalised DAR counts ")+
  theme_minimal(base_size=13)

  pmd<-pmd+scale_x_continuous(breaks=seq(-20,20,20),labels=c("-20K", "H3K27ac","20K"))
  
```
- Fig4. D Right (swapped DARs and H3K27ac)

```{r}
read_tsv("data_figures/DAR_aroundH3K27ac_Mes_SW_20K.txt")->dar_enh_sw


pmds<- ggplot(dar_enh_sw, aes(bin,line))+
  geom_line(color="blue",size=1) +
  ylim(0, 200)+
 #ggtile("DAR  normalised counts around H3K27ac") 
  xlab(" ")+
  ylab("normalised DAR counts,swapped ")+
  theme_minimal(base_size=13)

  pmds<-pmds+scale_x_continuous(breaks=seq(-20,20,20),labels=c("-20K", "H3K27ac","20K"))
   

```
-Fig4 D together
```{r}

 # penh<-grid.arrange(pmd, pmds, ncol=1)
penh<-grid.arrange(pmd, pmds, nrow=1)
penh

```
-Fig5A Left (DEG-Mes here)
```{r}
scatDEG <- read_tsv("data_figures/scatter_Mes_80K_data.txt")
#scatDEG
v1<-scatDEG$V1
v2<-scatDEG$V2
v3<-scatDEG$V3
dd1 <- data.frame(count=v1,line=v2)
#head(ds1)

dd2 <- data.frame(count=v1,line=v3)
#head(ds2)

#-----------------plot with superimposing
pd<- ggplot(dd1, aes(count,line))+
  geom_line(color="blue",size=2) +
  geom_point(data = dd2, color = "blue", size=2) +# re-define data and overwrite top layer inheritance
  #geom_point(data = df3, color = "green", size=3) +# re-define data and overwrite top layer inheritance
  xlim(0, 25)+
  #ggtitle("Correlation of frequency of diff accessible regions\n and average gene expression, r=0.07, TSS vicinity=80K \n Similarly High Expressed") +
  xlab("DAR counts") + ylab(" ")+
  ylab("average GE")+
  ylim(-3,6)+
  theme_classic2(base_size=14)
#pd
pd<-pd+scale_x_continuous(breaks=seq(0,20,10),labels=c("0", "10","20"))
pd

```
-Fig 5A middle (SEGs scatter)
```{r}
scatSEG <- read_tsv("data_figures/scatter_Sim_80K_data.txt")

#--------------------make two different data frames for re-defining on a plot further (to superimpose plots)

v1<-scatSEG$V1
v2<-scatSEG$V2
v3<-scatSEG$V3
ds1 <- data.frame(count=v1,line=v2)
#head(ds1)

ds2 <- data.frame(count=v1,line=v3)
#head(ds2)

#-----------------plot with superimposing
ps<- ggplot(ds1, aes(count,line))+
  geom_line(color="orange",size=2) +
  geom_point(data = ds2, color = "orange", size=2) +# re-define data and overwrite top layer inheritance
    xlim(0, 25)+
  #ggtitle("Correlation of frequency of similarly accessible regions\n and average gene expression, r=0.07, TSS vicinity=80K \n Similarly High Expressed") +
  xlab("SAR counts") + ylab(" ")+
  #ylab("gene expression")+
  ylim(-3,6)+
  theme_classic2(base_size=13)
ps
  ps<-ps+scale_x_continuous(breaks=seq(0,20,10),labels=c("0", "10","20"))
ps


```

-Figure5B
-fig5B Left
```{r}
read_tsv("C:/Users/ia1/Documents/R_work/Figures4JDA/data_figures/scatter_line_Mes_80K_H3K27ac_data2.txt")->dm1
read_tsv("C:/Users/ia1/Documents/R_work/Figures4JDA/data_figures/scatter_point_Mes_80K_H3K27ac_data2.txt")->dm2


ppd<- ggplot(dm1, aes(count,line))+
  geom_line(color="cyan",size=2) +
  geom_point(data = dm2, color = "cyan", size=2) +# re-define data and overwrite top layer inheritance
  
  xlim(0, 15)+
  #ggtitle("Correlation of frequency of similarly accessible regions\n and average gene expression, r=0.07, TSS vicinity=80K \n Similarly High Expressed") +
  xlab("H3K27ac counts") + ylab("average GE ")+
  theme_classic2(base_size=14)
#ppd

```
-Fig 5A left middle together
```{r}

pscat<-grid.arrange(pd, ps, nrow=1)
pscat

```
-Fig5A right
```{r}
read_tsv("data_figures/vic_corrEct_corrMes.txt")->vic_corr

vic_corr%>%
  mutate(set='DEG_DAR')%>%
  filter(layer=='Mes')->vic_corr1
  head(vic_corr1)
## # A tibble: 6 x 3
##   vicinity  corr layer
##      <dbl> <dbl> <chr>
## 1       10  0.53 Ect  
## 2       20  0.75 Ect  
## 3       40  0.84 Ect  
## 4       80  0.67 Ect  
## 5      120  0.6  Ect  
## 6      160  0.71 Ect
pc <- ggplot(vic_corr1, aes(x=vicinity, y=corr, colour=set)) + 
  geom_line(size=0.5,show.legend = FALSE)+
  scale_colour_manual(values=c("blue"))+

  theme_classic(base_size=14)+

  #ggtitle("Frequency of Differentially Chromatin-Accessible Regions \n in TSS vicinity (80K), DEG upregulated genes") +
  xlab("TSS vicinity") + ylab("correlation")+
 #geom_smooth(size=2, show.legend = FALSE)
 geom_smooth(size=2,show.legend = FALSE)
  
  pc<-pc+scale_x_continuous(breaks=seq(0,300,100),labels=c("TSS", "100K","200K","300K"))
  pc
  
#pc+theme(panel.background = element_rect(fill = "white", colour = "grey50"))
  grid.arrange(pd,ps, pc,nrow=1)

```


-Fig5B Right
-
```{r}
read_tsv("C:/Users/ia1/Documents/R_work/older_figs/fig3_Jan/vic_corrEct_corrMes_H3K27ac.txt")->vic_corr
head(vic_corr)

vic_corr%>%
  filter(layer=='Mes')->vic_corrM

pcc <- ggplot(vic_corrM, aes(x=vicinity, y=corr, colour=layer)) + 
  #geom_line(size=1)+
  scale_colour_manual(values=c("cyan","magenta"))+

  theme_classic2(base_size=15)+

  #ggtitle("Frequency of Differentially Chromatin-Accessible Regions \n in TSS vicinity (80K), Mesoderm upregulated genes") +
  xlab("TSS vicinity, KB") + ylab("correlation")+
  geom_smooth(size=2, show.legend = FALSE)

pcc<-pcc+scale_x_continuous(breaks=seq(0,300,100),labels=c("TSS", "100K","200K","300K"))
  
  #pcc-----------------together
  grid.arrange(ppd, pcc,nrow=1)

```

-Fig6A  DMR
```{r}
dmr_enh<-read_tsv("data_figures/DMR_end_H3k27ac_End_20K_data.txt")

pdmr_end_enh <- ggplot(dmr_enh, aes(x=bin, y=count)) + 
  geom_line(colour="cyan",size=1,show.legend = FALSE)+
  theme_classic(base_size=13)+
  xlab(" ") +
  ylab("DMR count")
 
  pdmr_end_enh<-pdmr_end_enh+scale_x_continuous(breaks=seq(-10000,10000,10000),labels=c("-10K", "H3K27ac","10K"))
 pdmr_end_enh

```
- DMR around DAR, Endoderm
```{r}

dmr_dar<-read_tsv("data_figures/DMR_end_DAR_End_20K_data.txt")
pdmr_end_dar <- ggplot(dmr_dar, aes(x=bin, y=count)) + 
  geom_line(colour="black",size=1,show.legend = FALSE)+
  theme_classic(base_size=13)+
  #xlab("H3K27ac vicinity") +
  ylab("DMR count")+
  xlab("")
 
  pdmr_end_dar<-pdmr_end_dar+scale_x_continuous(breaks=seq(-10000,10000,10000),labels=c("-10K", "DAR","10K"))
 #pdmr_end

```
-Fig6A together
```{r}

 
pdmr<-grid.arrange(pdmr_end_dar, pdmr_end_enh, nrow=1)
pdmr

```
- Ect En Mes Enh between
```{r}

read_excel("data_figures/motifs_F_ect.xlsx")->Li_ect
read_excel("data_figures/motifs_F_end.xlsx")->Li_end
read_excel("data_figures/motifs_F_mes.xlsx")->Li_mes

listik_a5_EcEnM2=c(Li_ect,Li_end,Li_mes)
#listik_w_dar_dmr_Ec2=c(vwec2,vdarec2,vdmrec2)
ur<-upset(fromList(listik_a5_EcEnM2),empty.intersections = "on")

#together

dmr_ur<-grid.arrange(pdmr, ur, nrow=1)
dmr_ur
```
-get distinct refined elements in final intersections
```{r}
library(VennDiagram)
overlapIntersect<-calculate.overlap(listik_a5_EcEnM2)
# intersection of Enh DAR DMR, a5
overlapIntersect
```
-KS test
```{r}

xy<-read_tsv("data_figures/frac_DEG_SEG_TAD.txt")
#xy

x <- xy$deg
y <- xy$seg
#x<-rnorm(10)

#y<-rnorm(10,2)

# Do x and y come from the same distribution?
#ks.test(x, y)
ks.test(x,y, alt='less')

plot(ecdf(x), xlim = range(c(x, y)))
plot(ecdf(y), add = TRUE, lty = "dashed")
t.test(x, y, alternative = "g")
wilcox.test(x, y, alternative = "g")
ks.test(x, y, alternative = "l")

```

